framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -969,7 +969,12 @@ public T findByUuidIncludingRemoved(final String uuid) {
     @Override
     @DB()
     public T findByIdIncludingRemoved(ID id) {
-        return findById(id, true, null);
+        if (_cache != null) {
+            final Element element = _cache.get(id);
+            return element == null ? findById(id, true, null) : (T)element.getObjectValue();
+        } else {
```

Comment:
This else clause is unnecessary due the return in preceding if clause.  Please remove it to simplify the code.